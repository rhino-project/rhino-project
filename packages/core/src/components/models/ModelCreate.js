import PropTypes from 'prop-types';

import { useGlobalOverrides } from 'rhino/hooks/overrides';
import ModelWrapper from 'rhino/components/models/ModelWrapper';
import ModelCreateBase from './ModelCreateBase';
import ModelCreateHeader from './ModelCreateHeader';
import ModelCreateForm from './ModelCreateForm';
import ModelCreateActions from './ModelCreateActions';

const defaultComponents = {
  ModelCreate: ModelCreateBase,
  ModelCreateHeader,
  ModelCreateForm,
  ModelCreateActions
};

const ModelCreate = ({ overrides, ...props }) => {
  // FIXME: Global overrides shouldn't be done this way
  const {
    ModelCreate,
    ModelCreateHeader,
    ModelCreateForm,
    ModelCreateActions
  } = useGlobalOverrides(defaultComponents, overrides, props);

  if (ModelCreateForm().props?.paths)
    console.warn('ModelCreateForm pass legacy paths prop');

  return (
    <ModelWrapper {...props} baseClassName="create">
      <ModelCreate paths={ModelCreateForm().props?.paths} {...props}>
        <ModelCreateHeader />
        <ModelCreateForm />
        <ModelCreateActions />
      </ModelCreate>
    </ModelWrapper>
  );
};

ModelCreate.propTypes = {
  model: PropTypes.object.isRequired,
  overrides: PropTypes.object
};

export default ModelCreate;
