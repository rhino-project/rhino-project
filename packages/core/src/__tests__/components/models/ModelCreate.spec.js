import ModelCreate from 'rhino/components/models/ModelCreate';
import { sharedModelTests } from './sharedModelTests';

/* eslint react/display-name: 0, react/prop-types: 0 */
describe('ModelCreate', () => {
  sharedModelTests(ModelCreate);
});
