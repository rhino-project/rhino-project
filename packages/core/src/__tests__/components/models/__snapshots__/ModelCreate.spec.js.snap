// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModelCreate > should allow local overrides 1`] = `
<DocumentFragment>
  <div
    class="model-create model-create-user"
  >
    <div>
      Foo
    </div>
    <div>
      Bar
    </div>
    <div>
      Baz
    </div>
  </div>
</DocumentFragment>
`;

exports[`ModelCreate > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelCreate > should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
